homeassistant/components/brunt/cover.py

```diff
@@ -100,37 +144,74 @@ def move_state(self) -> int | None:
-        self._attr_is_closed = self.current_cover_position == CLOSED_POSITION
-        self._attr_extra_state_attributes = {
-            ATTR_ATTRIBUTION: ATTRIBUTION,
-            ATTR_REQUEST_POSITION: self.request_cover_position,
-        }
+    @property
+    def is_opening(self) -> bool | None:
+        """Return if the cover is opening or not."""
+        return self.move_state == 1
+
+    @property
+    def is_closing(self) -> bool | None:
+        """Return if the cover is closing or not."""
+        return self.move_state == 2
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Return true if cover is closed, else False."""
+        return self.current_cover_position == CLOSED_POSITION
 
-    def open_cover(self, **kwargs):
+    async def async_open_cover(self, **kwargs) -> None:
         """Set the cover to the open position."""
-        self._bapi.changeRequestPosition(OPEN_POSITION, thingUri=self._thing_uri)
+        await self._async_update_cover(OPEN_POSITION)
 
-    def close_cover(self, **kwargs):
+    async def async_close_cover(self, **kwargs) -> None:
         """Set the cover to the closed position."""
-        self._bapi.changeRequestPosition(CLOSED_POSITION, thingUri=self._thing_uri)
+        await self._async_update_cover(CLOSED_POSITION)
 
-    def set_cover_position(self, **kwargs):
+    async def async_set_cover_position(self, **kwargs) -> None:
         """Set the cover to a specific position."""
-        self._bapi.changeRequestPosition(
-            kwargs[ATTR_POSITION], thingUri=self._thing_uri
-        )
+        await self._async_update_cover(kwargs[ATTR_POSITION])
+
+    async def _async_update_cover(self, position) -> None:
+        """Set the cover to the new position and wait for the update to be reflected."""
+        try:
+            await self._bapi.async_change_request_position(
+                position, thingUri=self._thing.thingUri
+            )
+        except ClientResponseError:
+            _LOGGER.warning("Unable to reposition %s", self._thing.NAME)
+            return
+        self.coordinator.update_interval = FAST_INTERVAL
+        await self.coordinator.async_request_refresh()
+
+    @callback
+    def _brunt_update_listener(self) -> None:
+        """Update the update interval after each refresh."""
+        if (
+            self.request_cover_position
+            == self._bapi.last_requested_positions[self._thing.thingUri]
+            and self.move_state == 0
+        ):
+            self.coordinator.update_interval = REGULAR_INTERVAL
+        else:
+            self.coordinator.update_interval = FAST_INTERVAL
+        assert self._attr_extra_state_attributes
+        self._attr_extra_state_attributes[
+            ATTR_REQUEST_POSITION
+        ] = self.request_cover_position
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device_info."""
+        assert self.unique_id
+        assert self.name
```

Comment:
Asserting should be avoided.
```suggestion
```

Reply:
These are needed for mypy checks, or do you prefer `# type: ignore` instead?

Reply:
No, ignoring is not the correct solution. Either check for them and handle the typing case, cast them or, use a different source.

For `"name"`, `self._thing.NAME` can be used instead.
For the identifier, you could store `serial` in the constructor to `self._serial` and use it here.

Reply:
clear, made a change to fix it